### PR TITLE
feat: add fromHex to MultisigRune

### LINKED ISSUE #733: make virtual multisig user proxying optional
Do we want to enable this within the pattern itself?

### PR DESCRIPTION
**Pull request by ryanleecode**
Fixes #733.

Basic copy and paste implementation from `VirtualMultisigRune` except parameters changed.

### CODE REVIEW

File: patterns/multisig/MultisigRune.ts [hunk 1]
```diff
@@ -22,6 +27,8 @@ export class MultisigRune<out C extends Chain, out U> extends PatternRune<Multis
   threshold = this.value.map(({ threshold, signatories }) => threshold ?? signatories.length - 1)
   accountId = Rune.fn(multisigAccountId).call(this.value.access("signatories"), this.threshold)
   address = MultiAddress.Id(this.accountId)
+  encoded = this.value.map((m) => $multisig.encode(m))
```
Comment (harrysolovay) [inline patterns/multisig/MultisigRune.ts:30]: `Rune.constant($multisig).into(CodecRune).encoded(this)`

File: patterns/multisig/MultisigRune.ts [hunk 2]
```diff
@@ -22,6 +37,8 @@ export class MultisigRune<out C extends Chain, out U> extends PatternRune<Multis
   threshold = this.value.map(({ threshold, signatories }) => threshold ?? signatories.length - 1)
   accountId = Rune.fn(multisigAccountId).call(this.value.access("signatories"), this.threshold)
   address = MultiAddress.Id(this.accountId)
+  encoded = Rune.constant($multisig).into(CodecRune).encoded(this.value)
```
Comment (harrysolovay) [inline patterns/multisig/MultisigRune.ts:40]: Let's extract this codec rune into a member `$multisig`, then utilize here and in the `fromHex` method below.

File: patterns/multisig/MultisigRune.ts [hunk 3]
```diff
@@ -37,7 +37,7 @@ export class MultisigRune<out C extends Chain, out U> extends PatternRune<Multis
   threshold = this.value.map(({ threshold, signatories }) => threshold ?? signatories.length - 1)
   accountId = Rune.fn(multisigAccountId).call(this.value.access("signatories"), this.threshold)
   address = MultiAddress.Id(this.accountId)
-  encoded = Rune.constant($multisig).into(CodecRune).encoded(this.value)
+  encoded = CodecRune.from($multisig).encoded(this.value)
```
[the remark this answers is not in the document]
Comment (ryanleecode) [reply patterns/multisig/MultisigRune.ts:40]: Give error

```
Argument of type 'this' is not assignable to parameter of type 'VirtualMultisig | Rune<VirtualMultisig, U<this>>'.
  Type 'VirtualMultisigRune<C, U>' is not assignable to type 'VirtualMultisig | Rune<VirtualMultisig, U<this>>'.
    Type 'VirtualMultisigRune<C, U>' is not assignable to type 'Rune<VirtualMultisig, U<this>>'.
      Type 'this' is not assignable to type 'Rune<VirtualMultisig, U<this>>'.
        Type 'VirtualMultisigRune<C, U>' is not assignable to type 'Rune<VirtualMultisig, U<this>>'.
          The types returned by '_prime(...)' are incompatible between these types.
            Type 'Run<VirtualMultisig, U>' is not assignable to type 'Run<VirtualMultisig, U<this>>'.
              Type 'U' is not assignable to type 'U<this>'.deno-ts(2345)
```

[the remark this answers is not in the document]
Comment (harrysolovay) [reply patterns/multisig/MultisigRune.ts:40]: Do `.encoded(this.as(VirtualMultisigRune))`

**Pull request merged**